Makefile

```diff
@@ -0,0 +1,69 @@
+NIX_RUN = nix-shell --run 
+BASH_RUN = bash -c 
+DOCKER_ORG = ponelat
+
+# Check if nix-shell exists, use that. Else use bash.
+# Nix is a package manager that will install all the packages needed,
+# else you'll need to manually ensure you have all dependencies (like docker, docker-compose, nodejs, caddy, etc)
+ifneq (, $(shell which nix-shell))
+RUN := $(NIX_RUN)
+else
+RUN := $(BASH_RUN)
+endif
+
+help: ## Prints help for targets with comments
+	@grep -E '^[a-zA-Z._-]+:.*?## .*$$' $(MAKEFILE_LIST) | \
+		sort | \
+		awk 'BEGIN {\
+				FS = ":.*?## "; \
+				print "If you are lost, read the README.md.\nList of tasks.\n";\
+			}; { \
+		    printf "\033[36m%-30s\033[0m %s\n", $$1, $$2; \
+			}'
+
+dependencies: ## Prints list of dependencies in the project (for non-nix users)
+	@grep -E '^ *[a-zA-Z._-]+ *.*?## .*$$' shell.nix | \
+		sort | \
+		awk 'BEGIN {FS = "## "; \
+			print "List of required cli tools.\n";\
+			}; { \
+			split($$2, descArr, "\\(optional\\)"); \
+			dep = $$1; \
+			desc = $$2; \
+			color = "35m"; \
+			if (length(descArr) > 1) {\
+			  dep = dep  " (optional)"; \
+			  desc = descArr[2]; \
+				color = "33m"; \
+			} \
+			gsub(/^[ \t]+/,"",dep); \
+			gsub(/^[ \t]+/,"",desc); \
+			printf "\033[%s%-30s\033[0m%s\n",color,dep,desc \
+		}'
+
+docker-build: ## Build docker image of ponelat/overlays-cli
+	$(RUN) "docker build -f Dockerfile -t ponelat/overlays-cli ."
```

Comment:
Ideally you would run
```
docker buildx build --platform ${PLATFORM} 
```
to build multi arch docker images like for example https://hub.docker.com/r/kubeshop/overlay-cli/tags

To achieve that you can do something similar to:
```
PLATFORM?=linux/amd64,linux/arm64
docker-build:
   docker buildx create --name overlay-cli
   docker buildx use overlay-cli
   docker buildx build --platform ${PLATFORM} -t kubeshop/overlay-cli:latest --push .
```

Also for this you'd have to enable GH action so the image is build every time there is a release tag or similar

Reply:
> Ideally you would run
> 
> ```
> docker buildx build --platform ${PLATFORM} 
> ```
> 
> to build multi arch docker images like for example https://hub.docker.com/r/kubeshop/overlay-cli/tags
> 
> To achieve that you can do something similar to:
> 
> ```
> PLATFORM?=linux/amd64,linux/arm64
> docker-build:
>    docker buildx create --name overlay-cli
>    docker buildx use overlay-cli
>    docker buildx build --platform ${PLATFORM} -t kubeshop/overlay-cli:latest --push .
> ```
> 
> Also for this you'd have to enable GH action so the image is build every time there is a release tag or similar

@ponelat, I'd be interested in this getting merged along with the comment suggested above.